nncf/quantization/algorithms/post_training/algorithm.py

```diff
@@ -188,28 +211,207 @@ def apply(
+        return model_transformer.transform(transformation_layout)
+
+    def _dfs_quantize_models(
+        self,
+        parent_model: TModel,
+        parent_graph: NNCFGraph,
+        parent_dataset: Dataset,
+        parent_statistic_points: Optional[StatisticPointsContainer],
+        parent_model_cnt: int,
+    ) -> Tuple[TModel, int]:
+        """
+
+        :param parent_model:
+        :param parent_graph:
+        :param parent_dataset:
+        :param parent_statistic_points:
+        :param parent_model_cnt:
+        :return:
+        """
+        if self._has_if_op(parent_graph, self._backend_entity.if_node_metatype):
+            model_transformer = factory.ModelTransformerFactory.create(parent_model)
+
+            global_model_cnt = parent_model_cnt
+            for if_node in parent_graph.get_nodes_by_metatypes([self._backend_entity.if_node_metatype]):
+                for if_submodel_condition in (True, False):
+                    model_cnt = global_model_cnt + 1
+                    child_model = self._extract_if_submodel(model_transformer, if_node, if_submodel_condition)
+                    child_model_input_names = self._backend_entity.get_if_subgraph_input_names(
+                        parent_model, if_node, if_submodel_condition
+                    )
+                    if_cond_input_name = self._backend_entity.get_if_cond_input_name(parent_model, if_node)
+                    parent_model_with_additional_outputs = self._add_outputs_before_if_node(
+                        model_transformer, parent_model, if_node
+                    )
+
+                    child_dataset = self._make_dataset_for_child_model(
+                        factory.EngineFactory.create(parent_model_with_additional_outputs),
+                        parent_dataset,
+                        if_cond_input_name,
+                        child_model_input_names,
+                        if_submodel_condition,
+                        model_cnt,
+                    )
+
+                    child_quantized_model, model_cnt = self._dfs_quantize_models(
+                        child_model, NNCFGraphFactory.create(child_model), child_dataset, None, model_cnt
+                    )
+                    global_model_cnt = model_cnt
+
+                    nncf_logger.info(f"Set quantized model number {model_cnt} to the original model")
+                    parent_model = self._update_if_submodel(
+                        model_transformer, if_node, if_submodel_condition, child_quantized_model
+                    )
+                    if self.intermediate_model_dir:
+                        nncf_logger.info(
+                            f"Save quantized model number {model_cnt} to dir {self.intermediate_model_dir}"
+                        )
+                        self._backend_entity.dump_submodel(
+                            child_quantized_model, self.intermediate_model_dir, if_node, if_submodel_condition
+                        )
```

Comment:
From my understanding, `intermediate_model_dir` is used for the whole model saving during AAQ. Not for the subgraph's dumping while quantization.

Reply:
I want to use it to have the ability to dump subgraphs, because it is convenient for debugging purposes.

Reply:
`intermediate_model_dir` is not for debugging purposes at all.
It was introduced to save the fully quantized model during the AAQ process and give the ability to evaluate the different model states.
Dumping the N subgraphs instead of the 1 model is not a good experience, IMO. Even the `BiasCorrection` doesn't have such option, but it's really useful to dump a subgraph in each step since it's a more complex algorithm.

Reply:
Removed